meta-refkit/lib/oeqa/selftest/secureboot.py

```diff
@@ -0,0 +1,162 @@
+
+        config = ''
+
+        if signing_key:
+            meta_refkit_base = get_bb_var('META_REFKIT_BASE')
+
+            config += 'REFKIT_DB_KEY = "%s/files/secureboot/%s.key"\n' % (meta_refkit_base, signing_key)
+            config += 'REFKIT_DB_CERT = "%s/files/secureboot/%s.crt"\n' % (meta_refkit_base, signing_key)
+            config += 'REFKIT_IMAGE_EXTRA_FEATURES_append = " secureboot"\n'
+            config += 'REFKIT_IMAGE_EXTRA_INSTALL_append = " efivar"\n'
+
+        print('Building %s' % self.test_image)
+
+        self.write_config(config)
+        bitbake(self.test_image)
+        self.remove_config(config)
+
+        # Some of the cases depend on the timeout to expire. Allow overrides
+        # so that we don't have to wait 1000s which is the default.
+        overrides = {
+            'TEST_QEMUBOOT_TIMEOUT': boot_timeout,
+            }
+
+        print('Booting %s' % self.test_image)
+
+        try:
+            with runqemu(self.test_image, ssh=False,
+                          runqemuparams='ovmf.secboot slirp',
+                          overrides=overrides,
+                          image_fstype='wic') as qemu:
+
+                cmd  = 'modprobe efivarfs ; mount -t efivarfs none /sys/firmware/efi/efivars/'
+                cmd2 = 'SB_GUID_NAME=$(efivar -l|grep SecureBoot); efivar -d -n $SB_GUID_NAME'
+
+                status, output = qemu.run_serial(cmd)
+
+                self.assertTrue(status, 'Failed to mount efivarfs (status=%s):\n%s' % (status, output))
+
+                status, output = qemu.run_serial(cmd2)
+
+                # Test if the signed image has booted and the BIOS says UEFI
+                # Secure Boot is enabled.
+                self.assertTrue(status, 'SecureBoot not enabled (status=%s):\n%s' % (status, output))
+
+        except Exception:
+
+            # Currently runqemu() fails if 'login:' prompt is not seen and it's
+            # not possible to login as 'root'. Those conditions aren't met when
+            # booting to EFI shell. We catch the failure and parse the boot
+            # log to determine the success. Note: the timeout triggers verbose
+            # bb.error() but that's normal with some of the test cases.
+
+            workdir = get_bb_var('WORKDIR', self.test_image)
+            bootlog = "%s/testimage/qemu_boot_log" % workdir
+
+            with open(bootlog, "r") as log:
+
+                # This isn't right but all we can do at this point. The right
+                # approach would run commands in the EFI shell to determine
+                # the BIOS rejects unsigned and/or images signed with keys in
+                # dbx key store but that needs changes in oeqa framework.
+
+                output = log.read()
+
+                # PASS if the test case wants EFI shell and we see it in logs
+                if not (efishell and re.search('Shell>', output)):
+                    self.assertFail('The image did not boot properly. Boot log:\n%s' % output)
```

Comment:
Sorry, my comment (from memory) was wrong - the function is self.fail, not self.assertFail.